### ISSUE TITLE
There should be no source color highlighting if the label points to the whole line, or if the label is multiline

### ISSUE THREAD

**Issue by RDambrosio016** [brendanzab/codespan#261]
Codespan currently highlights in red whatever the primary label is pointing to in the source, while this looks great for errors inside of source code lines, it looks really ugly if the label points to the whole line. It ruins the point of highlighting, you can clearly see what the label is pointing at, a line of all red just looks like it does not belong there. In the case of multiline labels, usually they are used to point to blocks of source code, but this means if you highlight everything inside of them you end up with a giant blob of RED which just looks like it does not belong and it ruins the cleanliness of the diagnostic.

Labels should highlight their source code only if the label points to a subsection of the source code line, and should be fully disabled for multiline labels.

**Comment by brendanzab** [comment]
Yeah, this was definitely one of my original concerns about adding source highlighting - see https://github.com/brendanzab/codespan/issues/206#issuecomment-602284647 - maybe special casing full lines and multiline might be ok?

cc. @jonathandturner @dtolnay

**Comment by Johann150** [comment]
@RDambrosio016 What would you consider to be the "whole line"? All text (optionally without the leading whitespace). What about trailing whitespace?

**Comment by RDambrosio016** [comment]
@Johann150 the whole line is the line of source code will __All__ leading and trailing whitespace removed, as this isnt really viewable so there is no point in counting it in

**Comment by RDambrosio016** [comment]
Note that this also means that no highlighting should be done if the label goes over the actual source and into trailing or leading whitespace

**Comment by brendanzab** [comment]
One thing I'm pondering is whether it would be better to do this as a simpler configuration option. Like, personally I'm not the biggest fan of the source highlighting, but perhaps we could add in a boolean flag to enable it?

**Comment by Johann150** [comment]
Do you mean that this option should turn off all highlighting and just rely on the underlines?

**Comment by brendanzab** [comment]
Yeah - thinking of how rustc normally renders errors. Maybe an option like `highlight_source_text` or something 🤔

**Comment by Johann150** [comment]
I think that is not related to this issue
